fix: OTA should use develcoPrimarySwVersion for Develco devices

sjorge opened the pull request:
@Koenkk as mentioned on discord, we need to read develcoPrimarySwVersion for develco/frient devices instead of swBuildId as that is always returns null otherwise.

I found no clean way to check if we have the develcoPrimarySwVersion on genBasic or not.

I opted to check for the Develco/Frient manufacturerID and just read the right attribute, currently we are hitting the catch branch. If the attributes move from zh -> zhc in the future and a new develco device would be missing the modernExtend we'd still have the same situation a today. I think for now this is fine to do.

sjorge commented:
Right I somehow need to mock a Develco device for the tests, I'm pretty bad at that stuff.

sjorge commented:
Ok this isn't working, not sure how to get the test to hit line 141, override the manufacturerCode in the extra tests hit the added if, but I can't get it to hit the resturn at all.

Koenkk commented:
I'm wondering if we can push this down to the definition `meta` field, something like `versionAttributes`

sjorge committed:
fix: OTA loses version info for develco/frient

This fix is not perfect, as it will still print that it got updated from null to null.

We just moved the reconfigure later, which will trigger this https://github.com/Koenkk/zigbee-herdsman-converters/blob/000041cba216822dab151151c493b560678da91a/src/devices/develco.ts#L36
for develco/frient devices that have it in their configure. (Which should be all of them)

Of note though, ZCL marks both dateCode and swBuildId as optional attributes, so Develco/Frient returning UNSUPPORTED_ATTRIBUTE here is not outside of spec. Although I'm not sure there is a cleaner way to do this.

sjorge commented:
Reworked it based on our conversation on Discord. This is not a perfect fix, but it should at least not leave the device with a empty sw version.

Koenkk committed:
Update otaUpdate.test.js

Koenkk committed:
Update otaUpdate.ts

Koenkk commented:
Fixed the tests, thanks!
